.github/workflows/build-images.yaml

```diff
@@ -1,5 +1,10 @@
 name: "[Release] Build service images"
-on: [push, pull_request]
+on:
+    push:
+      branches: [ master ]
+    pull_request:
+      types: [opened, synchronize]
```

Comment:
what does `synchronize` mean? Images should be built on all branches push so they can be deployed to staging

Reply:
the synchronize event happen when the base branch is updated, it's the default values but turns out we don't need to set a value there so I will remove.
The CI should be the same, the only diff is that "on push" will only kick in on master. You can check the current CI, let me know if you find something weird

Reply:
I want building to happen on every push, even on branches so any changes can be deployed to staging

Reply:
okay so without creating a PR right? 
Because it would still be possible but with a PR

Reply:
I don't think it's possible looking at the doc and the various issues around it. 
The only alternatives I have found was to prefix the branch names eventually (or create a PR).

Do you think it's cumbersome? I have never used staging so not sure what is the usual process

Reply:
Right now, every push (regardless of the branch) triggers the CI. This behavior is intended so every single change on a PR branch for instance is tested, image built, etc...
If I understand this PR correctly, CI will only be trigger by push to master or the PR being opened but not when changes is pushed to a non-master branch (ex: addressing PR comments or wanting to deploy branch to staging).

Reply:
The original changes means, CI on:
- push to master
- push to any pull_request

Currently it triggers on "push to all branches and push to any pull_request" hence why we end up running the jobs twice. e.g: my last PR https://github.com/NangoHQ/nango/pull/1658
Which slows down our CI and increase our billing. 
<img width="935" alt="Screenshot 2024-02-12 at 15 56 57" src="https://github.com/NangoHQ/nango/assets/1637651/6b7dcd7f-4698-498c-85b8-8fff50dd11a9">

There is no way (afaik) to avoid this behavior without making a consensus on something else. 
I'm definitely open to suggestion, I was just proposing one consensus ☺️ 
Maybe @khaliqgant has an opinion on this?